### PR TITLE
Vercel FE Deployment

### PR DESCRIPTION
**Pull request by dabeeeenster**
Todo:

- [x] Reinstate removed /server files
- [x] Fix script running locally
- [x] Test slack
- [x] Remove generated project.js file

### CODE REVIEW

File: frontend/api/index.js [hunk 1]
```diff
@@ -43,18 +33,20 @@ app.get('/config/project-overrides', (req, res) => {
...
     const envToBool = (name, defaultVal) => {
         const envVar = `${process.env[name]}`;
         if (envVar === 'undefined') {
             return defaultVal;
         }
         return envVar === 'true' || envVar === '1';
     };
+    let sha = '';
+    /*
+    todo: fix?
```
Comment (kyle-ssg) [inline frontend/api/index.js]: Needs fixing?

File: frontend/api/index.js [hunk 2]
```diff
@@ -78,111 +69,73 @@ app.get('/config/project-overrides', (req, res) => {
...
     res.send(`window.projectOverrides = {
         ${output}
     };
     `);
 });
 
-// Optionally proxy the API to get around CSRF issues, exposing the API to the world
-// FLAGSMITH_PROXY_API_URL should end with the hostname and not /api/v1/
-// e.g. FLAGSMITH_PROXY_API_URL=http://api.flagsmith.com/
-if (process.env.FLAGSMITH_PROXY_API_URL) {
```
Comment (kyle-ssg) [inline frontend/api/index.js]: Isn't this still relevant?

File: .github/workflows/frontend-deploy-production.yml [hunk 3]
```diff
@@ -1,12 +1,13 @@
 name: Frontend Deploy to Production
 
-on:
-    push:
-        branches:
-            - main
-        paths:
-            - 'frontend/**'
-            - '.github/**'
+on: pull_request
```
Comment (kyle-ssg) [inline .github/workflows/frontend-deploy-production.yml]: Needs re-adding?

Comment (dabeeeenster) [reply .github/workflows/frontend-deploy-production.yml]: Done

File: frontend/api/index.js (hunk 2 above)
Comment (dabeeeenster) [reply frontend/api/index.js]: Yes good catch - reimplemented.

File: frontend/api/index.js (hunk 1 above)
Comment (dabeeeenster) [reply frontend/api/index.js]: We don't use this currently in prod

**Comment by dabeeeenster** [comment]
OK fixed up the remaining comments

File: .github/workflows/frontend-deploy-staging.yml [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 
-on:
-    push:
-        branches:
-            - main
-        paths:
-            - 'frontend/**'
-            - '.github/**'
+on: pull_request
+#todo: update to main
```
Comment (kyle-ssg) [inline .github/workflows/frontend-deploy-staging.yml]: Can remove comments

File: frontend/api/index.js [hunk 5]
```diff
@@ -43,18 +33,20 @@ app.get('/config/project-overrides', (req, res) => {
...
     const envToBool = (name, defaultVal) => {
         const envVar = `${process.env[name]}`;
         if (envVar === 'undefined') {
             return defaultVal;
         }
         return envVar === 'true' || envVar === '1';
     };
+    let sha = '';
+    /*
+    todo: implement across docker and vercel
```
Comment (kyle-ssg) [inline frontend/api/index.js:45]: Can remove comments

File: .github/workflows/frontend-deploy-staging.yml [hunk 6]
```diff
@@ -1,12 +1,13 @@
-name: Frontend Deploy to Staging
+name: Frontend Deploy to Vercel Staging
 
-on:
```
Comment (kyle-ssg) [inline .github/workflows/frontend-deploy-staging.yml:3]: Needs adding

**Pull request merged**